source/grid/manifold.cc

```diff
@@ -274,7 +336,8 @@ Point<spacedim>
 Manifold<dim, spacedim>::
 get_new_point_on_line (const typename Triangulation<dim, spacedim>::line_iterator &line) const
 {
-  return get_new_point (get_default_quadrature(line));
+  const std::pair<std::vector<Point<spacedim> >, std::vector<double> > points_weights(get_default_points_and_weights(line));
+  return get_new_point (points_weights.first,points_weights.second);
```

Comment:
Yes reducing this further requires some changes in the places calling these functions. I tried to keep things simple.